packages/angular/src/schematics/convert-tslint-to-eslint/convert.ts

```diff
@@ -0,0 +1,840 @@
+  updateObjPropAndRemoveDuplication,
+} from './lib/utils';
+import { ConvertTSLintToESLintSchema } from './schema';
+
+export default function (schema: ConvertTSLintToESLintSchema): Rule {
+  return (tree: Tree, context: SchematicContext) => {
+    /**
+     * We are not able to support --dry-run in this schematic, because we need to (potentially) both install
+     * and use the @nrwl/eslint-plugin-nx package within the same execution.
+     *
+     * This is a worthwhile trade-off to avoid the need to force the user to manually run an install before
+     * invoking the schematic, and the dry-run output doesn't offer a ton of value for this use-case anyway.
+     */
+    if (
+      process.argv.includes('--dry-run') ||
+      process.argv.includes('--dryRun')
+    ) {
+      context.logger.error(
+        'NOTE: This schematic does not support --dry-run. If you are running this in Nx Console, it should execute fine once you hit the "Run" button.\n'
+      );
+      return process.exit(1);
+    }
+
+    if (!tree.exists('package.json')) {
+      throw new Error(
+        'Could not find a `package.json` file at the root of your workspace'
+      );
+    }
+
+    /**
+     * Ensure that the @nrwl/eslint-plugin-nx is installed
+     */
+    const packageJSON = tree.read('package.json').toString('utf-8');
+    const json = JSON.parse(packageJSON);
+
+    if (
+      !json.devDependencies?.['@nrwl/eslint-plugin-nx'] &&
+      !json.dependencies?.['@nrwl/eslint-plugin-nx']
+    ) {
+      const version: string | undefined =
+        json.devDependencies['@nrwl/workspace'];
+
+      context.logger.info(
+        '\nINFO: We are installing @nrwl/eslint-plugin-nx at the same version as your other @nrwl dependencies so that we can use it as part of the config generation...\n'
+      );
+
+      const pmc = getPackageManagerCommand();
+      execSync(`${pmc.addDev} @nrwl/eslint-plugin-nx@${version}`, {
+        stdio: 'ignore', // suppress any warnings about missing peerDeps to avoid confusion, they are all installed as part of the remaining schematic execution
+      });
+      context.logger.info(
+        '      Successfully installed @nrwl/eslint-plugin-nx\n'
+      );
+    }
+
+    const projectConfig = getProjectConfig(tree, schema.project);
+    const projectRoot = projectConfig.root;
+
+    // Expected existing files
+    const rootTSLintJsonPath = join(normalize(tree.root.path), 'tslint.json');
+    const projectTSLintJsonPath = join(normalize(projectRoot), 'tslint.json');
+
+    /**
+     * May or may not exist yet, depending on whether or not the schematic has been run
+     * before, or if this is already a mixed ESLint + TSLint workspace
+     */
+    const rootESLintrcJsonPath = join(
+      normalize(tree.root.path),
+      '.eslintrc.json'
+    );
+
+    return chain([
+      /**
+       * Handle adding ESLint and related dependencies and configuration
+       */
+      addAllConfigFilesAndCoreDependencies(projectRoot),
```

Comment:
I don't know where the issue is but what I'm seeing on `nx-examples` is that the depConstraints didn't get migrated correctly.

I have `depConstraints` defined in `/tslint.json` but the `/.eslintrc.json` has the `depConstraints`.

Reply:
Please could you try and add a failing unit test to reproduce? The conversion logic for the nx-module-boundaries rule has its own tests in this PR

Reply:
Ok actually I had forgotten my unit tests were already baed on specifically that repo so looks like unit testing won't reproduce it. I'll try and reproduce with a local build on the repo itself

> I have depConstraints defined in /tslint.json but the /.eslintrc.json has the depConstraints.

I'm guessing this has a typo in it? And that the eslintrc does _not_ have them?

Reply:
Have reproduced locally on the nx-examples repo 👍 will dig into that issue

Reply:
Resolved